src/downsampler.cpp

```diff
@@ -17,16 +34,45 @@ void Downsampler::onInit()
   ros::NodeHandle& nh = getNodeHandle();
 
   sub_cloud_ = nh.subscribe<sensor_msgs::PointCloud2>("input_cloud", 1, &Downsampler::downsample_cloud_cb, this);
-  pub_downsampled_cloud_ = nh.advertise<pcl::PointCloud<pcl::PointXYZ> >("downsampled_points", 1);
-  pub_filtered_cloud_ = nh.advertise<pcl::PointCloud<pcl::PointXYZ> >("filtered_points", 1);
+  pub_downsampled_ = nh.advertise<pcl::PointCloud<pcl::PointXYZ> >("downsampled_points", 1);
+  pub_filtered_ = nh.advertise<pcl::PointCloud<pcl::PointXYZ> >("filtered_points", 1);
+  pub_ramp_ = nh.advertise<pcl::PointCloud<pcl::PointXYZ> >("ramp_points", 1);
+  pub_ground_ = nh.advertise<pcl::PointCloud<pcl::PointXYZ> >("ground_points", 1);
+  pub_ground_free_ = nh.advertise<pcl::PointCloud<pcl::PointXYZ> >("ground_free_points", 1);
+  pub_padding_ = nh.advertise<pcl::PointCloud<pcl::PointXYZ> >("padding_points", 1);
+
+  front_normal_.resize(4);
+
+  reconfigure_server_ = std::shared_ptr<dynamic_reconfigure::Server<DownsamplerConfig> >(
+      new dynamic_reconfigure::Server<DownsamplerConfig>(nh));
+  dynamic_reconfigure::Server<DownsamplerConfig>::CallbackType reconfigure_cb = boost::bind(&Downsampler::reconfigureCB,
+                                                                                            this, _1, _2);
+  reconfigure_server_->setCallback(reconfigure_cb);
 
   ros::NodeHandle& private_nh = getPrivateNodeHandle();
 
-  private_nh.param("min_range", min_range_, 0.0);
-  private_nh.param("max_range", max_range_, 100.0);
-  private_nh.param("leaf_size", leaf_size_, 0.03);
-  private_nh.param("filter_radius", filter_radius_, 0.03);
-  private_nh.param("min_points_threshold", min_points_threshold_, 3);
+  private_nh.param("min_range", min_range_, min_range_);
+  private_nh.param("max_range", max_range_, max_range_);
+  private_nh.param("leaf_size", leaf_size_, leaf_size_);
+  private_nh.param("filter_radius", filter_radius_, filter_radius_);
+  private_nh.param("min_points_threshold", min_points_threshold_, min_points_threshold_);
+
+  bool remove_ramp;
+  private_nh.param("remove_ramp", remove_ramp, remove_ramp_.load());
```

Comment:
It will load the default of the reconfigure cfg file but not overwrite them with the params from the parameter yaml